src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,471 @@
+﻿using System;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(pcFlyoutOpen)]
+    public class SplitButton : ContentControl, ICommandSource
+    {
+        private const string pcFlyoutOpen        = ":flyout-open";
+        protected const string pcChecked = ":checked";
+
+        /// <summary>
+        /// Raised when the user presses the primary part of the <see cref="SplitButton"/>.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> Click
+        {
+            add => AddHandler(ClickEvent, value);
+            remove => RemoveHandler(ClickEvent, value);
+        }
+
+        public static readonly RoutedEvent<RoutedEventArgs> ClickEvent = Button.ClickEvent;
```

Comment:
This doesn't feel right to me but OK. You also nuked the comment and I'll add back.

Reply:
Are you really sure this is correct? There is zero harm in just defining a new RoutedEvent yet there are clear concerns re-using Button.ClickEvent.

1. I couldn't find examples of doing this anywhere else in the code
2. WPF had an .AddOwner(), Avalonia does not
3. With the issue and double click events occurring this doesn't seem optimal anyway.

Reply:
Even I didn't know you could do that. I have to agree with robloo here, this seems odd and IMO RoutedEvent should get an `.AddOwner` like properties. It would align with the property system and WPF and is more intuitive. I just had a scan through the code base and it seems this is only used in `InputElement.cs` for the `Tapped` and `DoubleTapped` events from `Gestures` so it's kind of a hidden thing.

Reply:
I don't see why it's a "hidden thing", and why owner is important for routed events in Avalonia (it's literally used only in DevTools, where we actually would want to see Button for better grouping).
But I do agree with consistency concern. Opened issue https://github.com/AvaloniaUI/Avalonia/issues/7507

Reply:
New routed event registration was restored. But I will keep this conversation unresolved so link to it will work properly.